src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs

```diff
@@ -186,5 +186,99 @@ public void TestAddUsingForAddExtentionMethod11()
 @"using System ; using System . Collections ; using Ext2 ; class X : IEnumerable { public IEnumerator GetEnumerator ( ) { new X { { 1 , 2 , 3 } , { ""Four"" , ""Five"" , ""Six"" } , { '7' , '8' , '9' } } ; return null ; } } namespace Ext { static class Extensions { public static void Add ( this X x , int i ) { } } } namespace Ext2 { static class Extensions { public static void Add ( this X x , object [ ] i ) { } } } ",
 null, 1, true, true, null, false, null);
         }
+
+        [WorkItem(3818, "https://github.com/dotnet/roslyn/issues/3818")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddUsing)]
+        public void InExtensionMethodUnderConditionalAccessExpression()
+        {
+            var initialText =
+@"<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly"" CommonReferences=""true"">
+        <Document FilePath = ""Program"">
+namespace Sample
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string myString = ""Sample"";
+            var other = myString?[|.StringExtension()|].Substring(0);
+        }
+    }
+}
+       </Document>
+       <Document FilePath = ""Extensions"">
+namespace Sample.Extensions
+{
+    public static class StringExtensions
+    {
+        public static string StringExtension(this string s)
+        {
+            return ""Ok"";
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expectedText =
+@"using Sample.Extensions;
+namespace Sample
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string myString = ""Sample"";
+            var other = myString?.StringExtension().Substring(0);
+        }
+    }
+}";
+            Test(initialText, expectedText, isLine: false);
+        }
+
+        [WorkItem(3818, "https://github.com/dotnet/roslyn/issues/3818")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddUsing)]
+        public void InExtensionMethodUnderMultipleConditionalAccessExpressions()
+        {
+            var initialText =
+  @"<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly"" CommonReferences=""true"">
+        <Document FilePath = ""Program"">
+public class C
+{
+    public T F&lt;T&gt;(T x)
+    {
+        return F(new C())?.F(new C())?[|.Extn()|];
```

Comment:
@dpoeschl I understand (from previous experiences as well :smile: ) . the change here is simple - just reuse a helper which has been well tested. In theory we should be fine. That said, I find your `A?B.TEST?.C` interesting.. Let me add a test for that case :wink: